build/executorch-config.cmake

```diff
@@ -9,85 +9,117 @@
 # is:
 #
 # find_package(executorch REQUIRED)
-
+# -------
+#
+# Finds the ExecuTorch library
+#
+# This will define the following variables:
+#
+#   EXECUTORCH_FOUND        -- True if the system has the Torch library
+#   EXECUTORCH_INCLUDE_DIRS -- The include directories for torch
+#   EXECUTORCH_LIBRARIES    -- Libraries to link against
+#
 cmake_minimum_required(VERSION 3.19)
 
-set(_root "${CMAKE_CURRENT_LIST_DIR}/../..")
-set(required_lib_list executorch executorch_core portable_kernels)
-foreach(lib ${required_lib_list})
-  set(lib_var "LIB_${lib}")
-  add_library(${lib} STATIC IMPORTED)
-  find_library(
-    ${lib_var} ${lib}
-    HINTS "${_root}"
-    CMAKE_FIND_ROOT_PATH_BOTH
+# Find prebuilt libportable_lib.so. If found, assuming current file is inside
+# a pip package:
+#     <site-packages>/executorch/executorch-config.cmake.
+# If not found, assuming current file is inside cmake-out:
+#     <cmake-out>/cmake/ExecuTorch/executorch-config.cmake
+find_library(_portable_lib_LIBRARY _portable_lib.so PATHS "${CMAKE_CURRENT_LIST_DIR}/extension/pybindings/")
```

Comment:
I'd much rather have two configs: one for the repo, and one for the pip package. Right now, 99% of this file is inside the `if` or the `else` -- there's basically no logic shared between the two paths, so there's no real benefit to putting this logic in a single file. It also adds complexity and assumptions, like this comment describes: "if there's no .so, assume we're in the repo".

So then, you could leave the current executorch-config.cmake alone, and add a short new one like executorch-wheel-config.cmake with the wheel logic, and then rename it when copying it into the wheel.